lib/src/widgets/board_editor.dart

```diff
@@ -0,0 +1,188 @@
+                      child: Container(
+                        decoration: const BoxDecoration(
+                          color: Color(0x33000000),
+                          shape: BoxShape.circle,
+                        ),
+                      ),
+                    ),
+                  if (piece != null)
+                    Draggable(
+                      hitTestBehavior: HitTestBehavior.translucent,
+                      data: piece,
+                      feedback: PieceDragFeedback(
+                        piece: piece,
+                        squareSize: widget.squareSize,
+                        settings: widget.settings,
+                      ),
+                      childWhenDragging: const SizedBox.shrink(),
+                      onDragStarted: () => draggedPieceOrigin = squareId,
+                      onDraggableCanceled: (_, __) {
+                        widget.onDiscardedPiece?.call(squareId);
+                        draggedPieceOrigin = null;
+                      },
+                      child: PieceWidget(
+                        piece: piece,
+                        size: widget.squareSize,
+                        pieceAssets: widget.settings.pieceAssets,
+                      ),
+                    ),
+                ],
+              );
+            },
+            onAcceptWithDetails: (details) {
+              widget.onDroppedPiece?.call(
+                draggedPieceOrigin,
+                squareId,
+                details.data,
+              );
+              draggedPieceOrigin = null;
+            },
+          ),
+        ),
+      );
+    }).toList();
+
+    final background = widget.settings.enableCoordinates
+        ? widget.orientation == Side.white
+            ? widget.settings.colorScheme.whiteCoordBackground
+            : widget.settings.colorScheme.blackCoordBackground
+        : widget.settings.colorScheme.background;
+
+    return SizedBox.square(
+      dimension: widget.size,
+      child: Stack(
+        clipBehavior: Clip.none,
+        children: [
+          if (widget.settings.boxShadow.isNotEmpty ||
+              widget.settings.borderRadius != BorderRadius.zero)
+            Container(
+              clipBehavior: Clip.hardEdge,
+              decoration: BoxDecoration(
+                borderRadius: widget.settings.borderRadius,
+                boxShadow: widget.settings.boxShadow,
+              ),
+              child: background,
+            )
+          else
+            background,
+          ...pieceWidgets,
+        ],
+      ),
+    );
+  }
+}
+
+/// Scaled version of a piece that is currently being dragged.
+///
+/// You can use this to drag pieces onto a [BoardEditor] with the same appearance as when the pieces on the board are dragged.
+class PieceDragFeedback extends StatelessWidget {
+  const PieceDragFeedback({
+    super.key,
+    required this.piece,
+    required this.squareSize,
+    this.settings = const BoardEditorSettings(),
```

Comment:
In order to be used by both `Board` and `BoardEditor` it should not take a settings object but directly
```dart
...
    this.size = 2.0,
    this.offset = const Offset(0.0, -1.0),
...
```

Board an board editor settings objects will be passed to this widget so they can override these defaults.